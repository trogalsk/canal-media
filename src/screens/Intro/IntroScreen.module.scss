@import "../../styles/variables";

.Container {
  width: 100%;
  height: 100vh;
  background-image: url("/background.png");
  background-repeat: no-repeat;
  background-size: cover;
}

.Content {
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: $primary-text;
  text-align: center;
}

.Info {
  position: relative;
}

.Gradient {
  background: rgba(0, 0, 0, 0.4);
  background-image: linear-gradient(
    0deg,
    rgba(0, 0, 0, 0.8) 0,
    transparent 60%,
    rgba(0, 0, 0, 0.8)
  );
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
}

.Logo {
  position: absolute;
  left: 50%;
  transform: translate(-50%);
  top: 5%;
}

.H3 {
  margin-top: 2rem;
}

.Button {
  color: $primary-text;
  margin-top: 2rem;
  display: inline-block;
  background-color: $primary;
  padding: 0.8125rem 2.5rem;
  font-size: 1.9rem;
  font-weight: 700;
  border-radius: 8px;
  border-color: $primary;
}
