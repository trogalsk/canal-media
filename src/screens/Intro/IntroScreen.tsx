/*
 * @author BSG <dev@bsgroup.eu>
 * @copyright Better Software Group S.A.
 * @version: 1.0
 */
import React from "react";
import Link from "next/link";
import { ROUTES } from "../..//constants";

import styles from "./IntroScreen.module.scss";

export const IntroScreen = () => {
  return (
    <div className={styles.Container}>
      <div className={styles.Gradient}></div>
      <img className={styles.Logo} src="/logo.png" alt="Canal media" />
      <div className={styles.Content}>
        <div className={styles.Info}>
          <h1>Unlimited movies, TV shows, and more.</h1>
          <h2>Watch anywhere. Cancel anytime.</h2>
          <h3 className={styles.H3}>Ready to watch?</h3>
          <Link href={ROUTES.HOME} passHref>
            <button className={styles.Button}>Get started</button>
          </Link>
        </div>
      </div>
    </div>
  );
};
