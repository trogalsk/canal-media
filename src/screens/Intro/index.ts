/*
 * @author BSG <dev@bsgroup.eu>
 * @copyright Better Software Group S.A.
 * @version: 1.0
 */
import { IntroScreen as IntroScreenDefinition } from "./IntroScreen";

export const IntroScreen = IntroScreenDefinition;
