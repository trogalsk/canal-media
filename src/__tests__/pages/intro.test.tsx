import { render, screen } from '@testing-library/react'
import Intro from "../../pages/index";

describe('Intro', () => {
  it('renders a heading', () => {
    render(<Intro />)

    const heading = screen.getByRole('heading', {
      name: "Unlimited movies, TV shows, and more."
    })

    expect(heading).toBeInTheDocument()
  })

  it('renders a get started button', () => {
    render(<Intro />)

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent('Get started')
  })

  it('renders intro unchanged', () => {
    const { container } = render(<Intro />)
    expect(container).toMatchSnapshot()
  })
})
