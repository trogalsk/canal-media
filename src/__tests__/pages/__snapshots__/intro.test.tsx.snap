// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders intro unchanged 1`] = `
<div>
  <div
    class="Container"
  >
    <div
      class="Gradient"
    />
    <img
      alt="Canal media"
      class="Logo"
      src="/logo.png"
    />
    <div
      class="Content"
    >
      <div
        class="Info"
      >
        <h1>
          Unlimited movies, TV shows, and more.
        </h1>
        <h2>
          Watch anywhere. Cancel anytime.
        </h2>
        <h3
          class="H3"
        >
          Ready to watch?
        </h3>
        <button
          class="Button"
          href="/home"
        >
          Get started
        </button>
      </div>
    </div>
  </div>
</div>
`;
